tests/PactContinuationSpec.hs

```diff
@@ -0,0 +1,402 @@
+ 
+  allResults `shouldMatch` allChecks
+
+
+testRollbackErr :: Expectation
+testRollbackErr = do
+  let moduleName = "testRollbackErr"
+  adminKeys <- genKeys
+
+  moduleCmd        <- mkExec (T.unpack (pactWithRollbackErrCode moduleName))
+                      (object ["admin-keyset" .= [(_kpPublic adminKeys)]])
+                      Nothing [adminKeys] (Just "test1")
+  executePactCmd   <- mkExec ("(" ++ moduleName ++ ".tester)")
+                      Null Nothing [adminKeys] (Just "test2")
+  contNextStepCmd  <- mkCont (TxId 1) 1 False Null Nothing [adminKeys] (Just "test3")
+  rollbackErrCmd   <- mkCont (TxId 1) 1 True Null Nothing [adminKeys] (Just "test4")
+  checkStateCmd    <- mkCont (TxId 1) 2 False Null Nothing [adminKeys] (Just "test5")
+  allResults       <- runAll [moduleCmd, executePactCmd, contNextStepCmd,
+                              rollbackErrCmd, checkStateCmd]
+  flushDb
+
+  let moduleCheck        = makeCheck moduleCmd False Nothing
+      executePactCheck   = makeCheck executePactCmd False $ Just "step 0"
+      contNextStepCheck  = makeCheck contNextStepCmd False $ Just "step 1"
+      rollbackErrCheck   = makeCheck rollbackErrCmd True Nothing
+      checkStateCheck    = makeCheck checkStateCmd False $ Just "step 2"
+      allChecks          = [moduleCheck, executePactCheck, contNextStepCheck,
+                            rollbackErrCheck, checkStateCheck]
+ 
+  allResults `shouldMatch` allChecks
+
+
+testNoRollbackFunc :: Expectation
+testNoRollbackFunc = do
+  let moduleName = "testNoRollbackFunc"
+  adminKeys <- genKeys
+
+  moduleCmd        <- mkExec (T.unpack (threeStepPactCode moduleName))
+                      (object ["admin-keyset" .= [(_kpPublic adminKeys)]])
+                      Nothing [adminKeys] (Just "test1")
+  executePactCmd   <- mkExec ("(" ++ moduleName ++ ".tester)")
+                      Null Nothing [adminKeys] (Just "test2")
+  contNextStepCmd  <- mkCont (TxId 1) 1 False Null Nothing [adminKeys] (Just "test3")
+  noRollbackCmd    <- mkCont (TxId 1) 1 True Null Nothing [adminKeys] (Just "test4")
+  checkStateCmd    <- mkCont (TxId 1) 2 False Null Nothing [adminKeys] (Just "test5")
+  allResults       <- runAll [moduleCmd, executePactCmd, contNextStepCmd,
+                              noRollbackCmd, checkStateCmd]
+
+  let moduleCheck        = makeCheck moduleCmd False Nothing
+      executePactCheck   = makeCheck executePactCmd False $ Just "step 0"
+      contNextStepCheck  = makeCheck contNextStepCmd False $ Just "step 1"
+      noRollbackCheck    = makeCheck noRollbackCmd False $ Just "No rollback on step 1" -- not a failure
+      checkStateCheck    = makeCheck checkStateCmd True
+                           (Just "applyContinuation: txid not found: 1")
+      allChecks          = [moduleCheck, executePactCheck, contNextStepCheck,
+                            noRollbackCheck, checkStateCheck]
+
+  allResults `shouldMatch` allChecks
```

Comment:
This is a bad practice with tests. Perform individual asserts. This applies everywhere there's the "allResults `shouldMatch` allChecks

Reply:
We'll address this later